core/src/main/java/io/grpc/internal/AbstractServerImplBuilder.java

```diff
@@ -228,13 +242,26 @@ public Server build() {
     ArrayList<ServerStreamTracer.Factory> tracerFactories =
         new ArrayList<ServerStreamTracer.Factory>();
     if (statsEnabled) {
-      StatsContextFactory statsFactory =
-          this.statsFactory != null ? this.statsFactory : Stats.getStatsContextFactory();
-      if (statsFactory != null) {
-        CensusStatsModule censusStats =
-            new CensusStatsModule(statsFactory, GrpcUtil.STOPWATCH_SUPPLIER, true, recordStats);
-        tracerFactories.add(censusStats.getServerTracerFactory());
-      }
+      Tagger tagger = this.tagger != null ? this.tagger : Tags.getTagger();
```

Comment:
Just curious, are the three components coupled? For example, does one Tagger implementation only works with a certain StatsRecorder, or can they mix and match?

Reply:
The Tagger and TagContextBinarySerializer objects are coupled.  The StatsRecorder is independent, except that it also uses the gRPC Context key that is used by the tagging classes.